### ISSUE TITLE
[Bug]: False positive AA0073 for temporary table [NextMajor]

### ISSUE THREAD

**Issue by ralf-escher** [ALCops/Analyzers#335]
### Analyzer

ApplicationCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

AA0073

### Description

While running AL-Go for NextMajor, I get conflicting AL CodeCop warnings:

Using a temporary table variable without the Temp prefix triggers:
`AA0073: The name of temporary variable ErrorList must be prefixed with Temp.`

Using the Temp prefix triggers:
`AA0237: The name of non-temporary variable TempErrorList must not be prefixed with Temp.`

Do variables based on a table with `TableType = Temporary` still need to be declared as temporary record variables for naming rules?

This issue **only** occurs in the NextMajor run (BC29).

### AL code to reproduce

```al
                    var
                        ErrorList: Record "Package Structure Rule Error";

-->

                    var
                        TempErrorList: Record "Package Structure Rule Error" temporary;
```

**Comment by Arthurvdv** [comment]
Thanks for reporting this, @ralf-escher. AA0073 and AA0237 are both rules from Microsoft's CodeCop, not from ALCops. This project doesn't own or maintain those diagnostics.

If the issue persists, please report it at https://github.com/Microsoft/AL/ so the Microsoft AL team can look into it.

Closing as out of scope.

**Comment by ralf-escher** [comment]
https://github.com/microsoft/AL/issues/7811
